src/app/file-manager/file-explorer/file-explorer.component.ts

```diff
@@ -24,13 +24,16 @@ export class FileExplorerComponent implements OnInit {
   msgs: Message[];
 
   @Output('done') done: EventEmitter<any> = new EventEmitter();
+  @Output('count') count: EventEmitter<any> = new EventEmitter();
 
   files: TreeNode[];
   workspaces: any[];
 
-  selectedFiles: TreeNode[];
+  selectedCount = 0;
+  selectedFiles: TreeNode[] = [];
   selectedFile: TreeNode;
 
+  folderCount = 0;
   fileCount = 0;
   fileSize = 0;
```

Comment:
This code has many unused variables, such as `workspaces`, `selectedCount`, `folderCount`, `fileSize`. The constructor accepts `gcsService` and `firecloudService` but they are both unused as well. Unused variables make the code harder to read and maintain, so they should be deleted if they're not used.

Reply:
Unused variables have been removed